code/ui/blocks/src/blocks/SourceContainer.tsx

```diff
@@ -21,24 +29,51 @@ export interface SourceContextProps {
 
 export const SourceContext: Context<SourceContextProps> = createContext({ sources: {} });
 
+type SnippetRenderedEvent = {
+  id: StoryId;
+  source: string;
+  args?: Args;
+  format?: SyntaxHighlighterFormatTypes;
+};
+
+export const unknownArgsHash = '--unknown--';
```

Comment:
nit: this looks like a constant.

```suggestion
export const UNKNOWN_ARGS_HASH = '--unknown--';
```